spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/BasicEurekaRestTemplateFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.netflix.eureka.http;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.springframework.http.client.support.BasicAuthorizationInterceptor;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * @author Daniel Lavoie
+ */
+public class BasicEurekaRestTemplateFactory implements EurekaRestTemplateFactory {
+	@Override
+	public RestTemplate newRestTemplate(String serviceUrl) {
+		RestTemplate restTemplate = new RestTemplate();
```

Comment:
Shouldnt we use `RestTemplateBuilder`?

Reply:
I don't have any strong opinion on that. Pushed a new version with the builder.